packages/react-flicking/src/react-flicking/Flicking.tsx

```diff
@@ -0,0 +1,117 @@
+  // life cycle
+  public render() {
+    const props = this.props;
+    // tslint:disable-next-line:naming-convention
+    const Tag = props.tag as any;
+    const classPrefix = props.classPrefix;
+    const options = this.options;
+    const attributes: { [key: string]: any } = {};
+
+    for (const name in props) {
+      if (name in DEFAULT_OPTIONS) {
+        options[name] = props[name];
+      } else if (!(name in FLICKING_PROPS)) {
+        attributes[name] = props[name];
+      }
+    }
+    return (
+      <Tag {...attributes}>
+        <div className={`${classPrefix}-viewport`}>
+          <div className={`${classPrefix}-camera`}>
+            <ChildrenDiffer onUpdate={this.onUpdate}>
+              {this.renderPanels()}
+            </ChildrenDiffer>
+          </div>
+        </div>
+      </Tag>
+    );
+  }
+  public onUpdate = (result: ChildrenDiffResult<HTMLElement>) => {
+    if (typeof this.props.lastIndex === "number") {
+      this.setLastIndex(this.props.lastIndex!);
+    }
+    this.flicking!.sync(result as ChildrenDiffResult<HTMLElement>);
+    this.checkPlugins();
+    this.checkCloneCount();
+  }
+  public componentDidMount() {
+    this.flicking = new NativeFlicking(
+      findDOMNode(this) as HTMLElement,
+      this.options,
+    ).on({
+      moveStart: (e: FlickingEvent) => this.props.onMoveStart!(e),
+      move: (e: FlickingEvent) => this.props.onMove!(e),
+      moveEnd: (e: FlickingEvent) => this.props.onMoveEnd!(e),
+      holdStart: (e: FlickingEvent) => this.props.onHoldStart!(e),
+      holdEnd: (e: FlickingEvent) => this.props.onHoldEnd!(e),
+      select: (e: FlickingEvent) => this.props.onSelect!(e),
+      needPanel: (e: NeedPanelEvent) => this.props.onNeedPanel!(e),
+      change: (e: FlickingEvent) => this.props.onChange!(e),
+      restore: (e: FlickingEvent) => this.props.onRestore!(e),
+    });
+
+    if (this.props.status) {
+      this.setStatus(this.props.status);
+    }
+    this.checkPlugins();
+    this.checkCloneCount();
+  }
+  public componentWillUnmount() {
+    this.destroy();
+  }
+  // private
+  private checkPlugins() {
+    const { list, added, removed, prevList } = this.pluginsDiffer.update(this.props.plugins!);
+
+    this.addPlugins(added.map(index => list[index]));
+    this.removePlugins(removed.map(index => prevList[index]));
+  }
+  private checkCloneCount() {
+    const cloneCount = this.flicking!.getCloneCount();
+
+    if (this.state.cloneCount !== cloneCount) {
+      this.setState({
+        cloneCount,
+      });
+    }
+  }
+  private renderPanels() {
+    const length = this.state.cloneCount;
+    const children = React.Children.toArray(this.props.children) as Array<React.ReactElement<any>>;
+    let arr: Array<React.ReactElement<any>> = [...children];
+
+    for (let i = 0; i < length; ++i) {
+      arr = arr.concat(children.map(el => {
+        return <CloneComponent key={`clone${i}${el.key}`}>{el}</CloneComponent>;
```

Comment:
@daybrush Could you give me a reason why wrapping `el` with `CloneComponent` not by mapping `key` to `el` directly?

Reply:
`el` is also an original component and read-only and cannot be modified. Therefore, it creates a clone component. The key should not be the same because it should be unique to the parent.